src/main/scala/com/databricks/labs/overwatch/pipeline/Pipeline.scala

```diff
@@ -118,76 +118,128 @@ class Pipeline(
-      val expiredRecords = lastRunPricing
-        .filter('rnk === 1)
-        .withColumn("activeUntil",
-          when('activeUntil.isNull, date_sub(lit(pipelineSnapTime.asDTString).cast("date"), 1))
-            .otherwise('activeUntil)
-        )
-
-      // New pricing records ++ expired pricing records ++ historical records
-      val newCloudDetailsDF = newDBUPriceRecords
-        .unionByName(expiredRecords)
-        .unionByName(lastRunPricing.filter('rnk > 1))
+      // active costs to be expired due to replacement
+      val expiringRecords = lastRunDBUCosts
+        .filter(activeRecord&& 'contract_price =!= activeCost)
+        .withColumn("activeUntil", lit(pipelineSnapTime.asDTString).cast("date"))
+        .withColumn("isActive", lit(false))
+
+      // historical records ++ new pricing records ++ newly expired pricing records ++ unchanged active records
+      val updatedDBUCostDetail = newDBUPriceRecords
+        .unionByName(expiringRecords)
+        .unionByName(unchangedActiveCosts)
+        .unionByName(originalExpiredRecords)
         .drop("rnk")
-        .repartition(4)
+        .coalesce(1)
 
       // overwrite the target and preserve original Overwatch metadata
-      val overwriteTarget = cloudDetailTarget.copy(
+      val overwriteTarget = dbuCostDetail.copy(
         withOverwatchRunID = false,
         withCreateDate = false,
         _mode = WriteMode.overwrite
       )
-      database.write(newCloudDetailsDF, overwriteTarget, lit(null).cast("timestamp"))
+      database.write(updatedDBUCostDetail, overwriteTarget, lit(null).cast("timestamp"))
+    }
+
+  }
+
+  private def publishDBUCostDetails(): Unit = {
+    val costDetailDF = Seq(
+      DBUCostDetail(config.organizationId, "interactive", config.contractInteractiveDBUPrice, primordialTime.asLocalDateTime.toLocalDate, None, true),
+      DBUCostDetail(config.organizationId, "automated", config.contractAutomatedDBUPrice, primordialTime.asLocalDateTime.toLocalDate, None, true),
+      DBUCostDetail(config.organizationId, "sqlCompute", config.contractSQLComputeDBUPrice, primordialTime.asLocalDateTime.toLocalDate, None, true),
+      DBUCostDetail(config.organizationId, "jobsLight", config.contractJobsLightDBUPrice, primordialTime.asLocalDateTime.toLocalDate, None, true),
+    ).toDF().coalesce(1)
+
+    database.write(costDetailDF, BronzeTargets.dbuCostDetail, pipelineSnapTime.asColumnTS)
+    if (config.databaseName != config.consumerDatabaseName) BronzeTargets.dbuCostDetailViewTarget.publish("*")
+  }
+
+  private def publishInstanceDetails(): Unit = {
+    val logMsg = "instanceDetails does not exist and/or does not contain data for this workspace. BUILDING/APPENDING"
+    logger.log(Level.INFO, logMsg)
+    if (getConfig.debugFlag) println(logMsg)
+    val instanceDetailsDF = config.cloudProvider match {
+      case "aws" =>
+        InitializerFunctions.loadLocalCSVResource(spark, "/AWS_Instance_Details.csv")
```

Comment:
just general though - maybe we should host these CSV files somewhere? Or even take directly from Github, so people won't rely onto deployment of the new jars

Reply:
they are in the resources folder. I get your point but I don't have a great place to store them and make them programmatically available. Now that repo is public, yes, we could stream over https...just haven't done that. Kind of really low on my list. Also, these are just for first-run inits...customers will customize these for their costs so it's mainly for the structure and getting them the 80% of the way.